Orbs address scheme next version: network ID, virtual chain ID, encoding in parts

lbeder opened the pull request:
This PR implements the next Orbs address scheme, combining the spec from architecture. 

For example:

```bash
Public Key: 8d41d055d00459be37f749da2caf87bd4ced6fafa335b1f2142e0f44501b2c65
Virtual Chain ID: 640ed3
Network ID: 14 (M)

SHA256: 40784b5b15e6bb364263dbb598f262bc5c5b4c18a34806ca70be180c3d995e0d
Account ID: RIPEMD160 of SHA256: c13052d8208230a58ab363708c08e78f1125f488
Network ID + Version + Virtual Chain + Account ID: 1400640ed3c13052d8208230a58ab363708c08e78f1125f488
Checksum: 61f04bfc
Raw public address: 1400640ed3c13052d8208230a58ab363708c08e78f1125f48861f04bfc

BASE58 (in parts): M1EXMPnnaWFqRyVxWdhYCgGzpnaL4qBy4QFsJu1
```

lbeder commented:
@idandash @netoneko this PR is good to go.

idandash commented on client/crypto-sdk/lib/address.cpp:
```diff
@@ -14,18 +15,34 @@
 const uint32_t Address::ACCOUNT_ID_SIZE = 20;
 const uint32_t Address::CHECKSUM_SIZE = 4;
-const uint32_t Address::ADDRESS_LENGTH = 34;
+const uint32_t Address::ADDRESS_LENGTH = 39;
```
Please comment on how you got to these numbers to keep it maintainable

lbeder replied:
Done

idandash commented on client/crypto-sdk/lib/address.cpp:
```diff
@@ -14,18 +15,34 @@
+Address::Address(const string &publicKey, const string &virtualChainId, const std::string &networkId) :
+    publicKey_(Utils::Hex2Vec(publicKey)), virtualChainId_(Utils::Hex2Vec(virtualChainId))  {
+
+    vector<uint8_t> decodedNetworkId = Base58::Decode(networkId);
```
I find this decoding an overkill

idandash replied:
Not fixed yet

lbeder replied:
I've written below - this will be changed in the future, because: 1) Context switch; 2) It was done according to the spec; 3) There more pending changes and this one will take 2-4 hours to do, since all the test addressed will need to be regenerated.

idandash commented on client/crypto-sdk-javascript/src/address.cpp:
```diff
@@ -67,7 +73,19 @@ napi_value Address::New(napi_env env, napi_callback_info info) {
+        char virtualChainId[virtualChainIdLength + 1];
+        napi_get_value_string_utf8(env, args[1], virtualChainId, virtualChainIdLength + 1, 0);
+
+        size_t networkIdLength;
```
Assumes little-endianess. Please prevent compilation on big-endian environments.

lbeder replied:
Done

lbeder commented on client/crypto-sdk-javascript/src/address.cpp:
```diff
@@ -150,6 +206,38 @@ napi_value Address::GetChecksum(napi_env env, napi_callback_info info) {
+}
+
+napi_value Address::GetTestNetworkId(napi_env env, napi_callback_info info) {
+    napi_status status;
```
Done and done.

lbeder commented:
@idandash I've addressed your comments. I've noted of the extra encoding and perhaps change it in the future, once the mobile integration is complete (since it required to regenerate the addresses).

lbeder commented:
@idandash please approve and let's merge.
